File: backend/user-service/src/main/java/ru/maxizenit/backend/chatservice/repository/InterestRepository.java
package ru.maxizenit.backend.chatservice.repository;

import java.util.Collection;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import ru.maxizenit.backend.chatservice.entity.Interest;

public interface InterestRepository extends CrudRepository<Interest, Long> {

  @Query("select i from Interest i where i.id in :ids")
  Collection<Interest> findAllWhereIdIn(Collection<Long> ids);
}

File: src/main/kotlin/org/maxizenit/maxigram/profile/InterestRepository.kt
package org.maxizenit.maxigram.profile

import org.jooq.DSLContext
import org.maxizenit.maxigram.jooq.Tables.INTEREST
import org.springframework.stereotype.Repository

interface InterestRepository {
    fun findAll(): List<Interest>
    fun findByIds(ids: Collection<Long>): List<Interest>
}

@Repository
class JooqInterestRepository(private val dsl: DSLContext) : InterestRepository {

    override fun findAll(): List<Interest> =
        dsl.select(INTEREST.ID, INTEREST.NAME)
            .from(INTEREST)
            .orderBy(INTEREST.ID)
            .fetch { Interest(it.get(INTEREST.ID), it.get(INTEREST.NAME)) }

    override fun findByIds(ids: Collection<Long>): List<Interest> {
        if (ids.isEmpty()) return emptyList()
        return dsl.select(INTEREST.ID, INTEREST.NAME)
            .from(INTEREST)
            .where(INTEREST.ID.`in`(ids))
            .orderBy(INTEREST.ID)
            .fetch { Interest(it.get(INTEREST.ID), it.get(INTEREST.NAME)) }
    }
}
